docs/spec/resource-scopes.md

```diff
@@ -8,30 +8,56 @@ See [here][arm-scopes] for more information on ARM scopes.
 
 ## Declaring and using scopes
 
+### Declaring the target scope(s)
+Unless otherwise specified, Bicep will assume that a given `.bicep` file is to be deployed at a resource group scope, and will validate resources accordingly. If you wish to change this scope, or define a file that can be deployed at multiple scopes, you must use the `target` keyword with either a string or array value as follows:
+
+```bicep
+// this file can only be deployed at a subscription scope
+target = 'subscription'
+```
+
+```bicep
+// this file can be deployed at either a tenant or managementGroup scope
+target = [
+  'tenant'
+  'managementGroup'
+]
+```
+
+The following strings are permitted for the `target` keyword: `'tenant'`, `'managementGroup'`, `'subscription'`, `'resourceGroup'`, `'extension'`. Expressions are not permitted.
```

Comment:
what is `extension`?

Reply:
This relates to an extension resource - for example here we're deploying a resource of type `Microsoft.Insights/diagnosticSettings` as an extension of a resource of type `Microsoft.Cache/redis`: https://github.com/Azure/azure-quickstart-templates/blob/1940ad5ceae6e0750da9cc2d089f186c8344571f/101-redis-cache/azuredeploy.json#L95-L97

Reply:
But I don't understand how extension can be a target. IOW, there's no Deployment PUT API for extension resources.

Reply:
The PUT URI is something like this {resourceId}/providers/{extensionNamespace}/{extensionType}/{extensionName}. It isn't a deployment API - just a PUT API of the resource (just like with tracked or proxy resources at RG or Sub scopes).

Reply:
There's also cascade extensions that are extension resources on other extension resources 😈. 

Another interesting thing that comes to mind is do we need any additional constraints on extension resources? Most extension resources can only  extend a limited set of resources

Reply:
> But I don't understand how extension can be a target. IOW, there's no Deployment PUT API for extension resources.

Yeah.... That's a great point - I was not thinking properly here! it would be useful in theory, but not possible with the current approach of generating nested templates.